Code/GraphMol/Sgroup.cpp

```diff
@@ -0,0 +1,322 @@
+//
+//
+//  Copyright (C) 2002-2018 Greg Landrum and T5 Informatics GmbH
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include "Sgroup.h"
+#include "ROMol.h"
+
+namespace RDKit {
+
+static const std::vector<std::string> sGroupTypes = {
+    // polymer sgroups:
+    "SRU", "MON", "COP", "CRO", "GRA", "MOD", "MER", "ANY",
+    // formulations/mixtures:
+    "COM", "MIX", "FOR",
+    // other
+    "SUP", "MUL", "DAT", "GEN"};
+
+static const std::vector<std::string> sGroupSubtypes = {"ALT", "RAN", "BLO"};
+static const std::vector<std::string> sGroupConnectTypes = {"HH", "HT", "EU"};
+
+SGroup::SGroup(ROMol *owning_mol, const std::string &_type)
+    : d_id(0), d_type(_type), dp_mol(owning_mol) {
+  PRECONDITION(owning_mol, "owning molecule is nullptr");
+}
+
+void SGroup::setOwningMol(ROMol *mol) {
```

Comment:
It doesn't appear that this code is called.  The SGroup constructor must take an owning mol so this function is redundant.